core/src/main/java/net/sourceforge/jnlp/tools/JarCertVerifier.java

```diff
@@ -342,15 +347,48 @@ VerifyResult verifyJarEntryCerts(final String jarPath, final boolean jarHasManif
                 final Certificate cert = certPath.getCertificates().get(0);
                 if (cert instanceof X509Certificate) {
                     checkCertUsage(certPath, (X509Certificate) cert);
+
                     final ZonedDateTime notBefore = zonedDateTime(((X509Certificate) cert).getNotBefore());
                     final ZonedDateTime notAfter = zonedDateTime(((X509Certificate) cert).getNotAfter());
-                    if (now.isBefore(notBefore)) {
-                        certInfo.setNotYetValidCert();
-                    }
-                    if (now.isAfter(notAfter)) {
-                        certInfo.setHasExpiredCert();
-                    } else if (now.plus(6, MONTHS).isAfter(notAfter)) {
-                        certInfo.setHasExpiringCert();
+
+                    final Optional<Timestamp> optionalSignatureTimestamp = Optional.ofNullable(codeSigners.get(certPath))
+                            .map(CodeSigner::getTimestamp);
+                    final X509Certificate tsaCertificate = (X509Certificate) optionalSignatureTimestamp
+                            .map(Timestamp::getSignerCertPath)
+                            .map(CertPath::getCertificates)
+                            .map(certs -> certs.get(0))
+                            .filter(c -> c instanceof X509Certificate)
+                            .orElse(null);
+
+                    if (tsaCertificate != null) {
+                        final ZonedDateTime tsaNotBefore = zonedDateTime(tsaCertificate.getNotBefore());
+                        final ZonedDateTime tsaNotAfter = zonedDateTime(tsaCertificate.getNotAfter());
+                        final ZonedDateTime signedAt = zonedDateTime(optionalSignatureTimestamp.get().getTimestamp());
```

Comment:
We should use orElseThrow instead of get here. I understand that I can not happen now that the val is null but maybe in future based on refactoring. So maybe throw a specific exception + description here.